### DISCUSSION TITLE
Blocked queries mapped to blocklist or regex?

### DISCUSSION THREAD

**Discussion by mljdivemaster** [ZachChristensen28/pihole_dns_app#27] [Ideas]
It would be interesting to load each blocklist as a lookup in Splunk. So when you analyze blocked DNS queries, you can see which blocklist was contributing to either a false positive or if the blocklist was against malware. It can bring up a score of which blocklist is the most effective in your DNS traffic.

This information is not available in DNS logs itself, so the blocklist needs to be populated independently in Splunk. A newer version of PiHole, don't they store this information in an SQLite database? Maybe a small script in PiHole that is exporting this info to a file, which can be read into Splunk.

Thoughts?

**Comment by ZachChristensen28** [comment]
I agree. I've been exploring ways to do this for a little while now but I haven't found a good method yet. Looking forward to future releases, there are discussions that this may be one of the things that can be extracted through their HTTP API.

In the meantime, I am going to create a branch on the [TA-pihole_dns add-on](https://github.com/ZachChristensen28/TA-pihole_dns/tree/enhancement-21) to start experimenting with methods to make this work.

**Comment by ZachChristensen28** [comment]
switch to the `enhancement-21` branch on the TA-pihole_dns add-on and then the `fr-blocklists-21` branch on this app. A scripted input was added to ingest the lists in `/etc/pihole`. Let me know if you have issues setting it up. I will begin creating dashboards with statistics on how each blocklist is being used.
